learning_resources/serializers.py

```diff
@@ -36,6 +36,40 @@ class Meta:
         fields = ["id", "name"]
 
 
+class WriteableTopicsMixin(serializers.Serializer):
+    """Class for editable topics"""
+
+    topics = WriteableSerializerMethodField()
+
+    def validate_topics(self, topics):
+        """Validator for topics"""  # noqa: D401
+        if len(topics) > 0:
+            if isinstance(topics[0], dict):
+                topics = [topic["id"] for topic in topics]
+            try:
+                valid_topic_ids = set(
+                    models.LearningResourceTopic.objects.filter(
+                        id__in=topics
+                    ).values_list("id", flat=True)
+                )
+            except ValueError as ve:
+                msg = "Topic ids must be integers"
+                raise ValidationError(msg) from ve
+            missing = set(topics).difference(valid_topic_ids)
+            if missing:
+                msg = f"Invalid topic ids: {missing}"
+                raise ValidationError(msg)
+        return {"topics": topics}
+
+    @extend_schema_field(LearningResourceTopicSerializer(many=True, allow_null=True))
```

Comment:
```suggestion
    @extend_schema_field(LearningResourceTopicSerializer(many=True))
```

I **think** we do not want `allow_null=True` here.

- When I try PATCHing http://localhost:8063/api/v1/userlists/1/ with `{ "topics": null }`, I get a validation error. So it seems like requests should not have null topics.
- `get_topics()` always returns a list and as far as I can tell responses always have listified topics (possibly empty).

Note: If we remove `allow_null=True`, drf-spectacular's OpenAPI will drop `nullable: true`, but will _not_ make `topics` required on requests.